src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1809,7 +1809,7 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
         } else {
 	  tempBlock.cacheObj.objName = "_:null_" + block.cacheObj.objName;
 	  ret = blockDir->get(dpp, &tempBlock, y); // check if null delete marker block already exists; if so, delete
-	  if (ret == 0) {
+	  if (ret == 0 && tempBlock.deleteMarker) {
```

Comment:
I do not have the latest changes of delete_obj so this change didnt make it in the cherry-picked commit, You can make a note of this and add it to your final delete_obj commit.